sesame/src/main/scala/org/w3/banana/sesame/io/SesameRDFReader.scala

```diff
@@ -75,3 +75,62 @@ class SesameJSONLDReader(implicit val ops: RDFOps[Sesame]) extends AbstractSesam
   def getParser() = new SesameJSONLDParser
 }
 
+
+
+class SesameRDFLoader(implicit val ops: RDFOps[Sesame]) extends RDFLoader[Sesame, Try] {
+   import org.openrdf.rio.Rio
+   import java.net.URL
+   import java.net.HttpURLConnection
+   import java.net.URLConnection
+
+  def load(url: URL): Try[Sesame#Graph] = Try {
+    val connection = getConnection(url)
+    val parser = getParser(url, connection)
+    val triples = new LinkedList[Statement]
+    val collector = new org.openrdf.rio.helpers.StatementCollector(triples) with CollectorFix {
+      val ops: RDFOps[Sesame] = SesameRDFLoader.this.ops
+    }
+    parser.setRDFHandler(collector)
+    val in: InputStream = connection.getInputStream
+    parser.parse(in, url.toString)
+    new LinkedHashModel(triples)
+  }
+
+  private def getConnection(url: URL): URLConnection = {
+      val connection = url.openConnection()
+      connection match {
+        case connection: HttpURLConnection =>
+        val acceptHeaderTurtlePriority =
```

Comment:
I wonder if this could be better achieved with a shapless HList (or similar) passed in as the third type parameter to RDFLoader? As well as the resulting signature being in common with the other readers, it would remove the hard coding of the priority and would be open for new reader types without modification?

Interestingly, that implementation would be richer that than the standard jena one.